packages/alfa-rules/src/sia-r15/rule.ts

```diff
@@ -105,3 +106,29 @@ export namespace Outcomes {
     )
   );
 }
+
+type Source = "srcdoc" | "src" | "invalid" | "nothing";
+
+/**
+ * @see https://html.spec.whatwg.org/multipage/iframe-embed-object.html#process-the-iframe-attributes
+ */
+function embeddedResource(iframe: Element): Map<Source, string> {
+  // srcdoc takes precedence.
+  if (iframe.attribute("srcdoc").isSome()) {
+    return Map.of(["srcdoc", iframe.attribute("srcdoc").get().value]);
+  }
+
+  // Otherwise, grab the src attribute.
+  function getUrl(value: string): Map<Source, string> {
+    try {
+      return Map.of(["src", new URL(value).href]);
+    } catch {
+      return Map.of(["invalid", ""]);
+    }
+  }
+
+  return iframe
+    .attribute("src")
+    .map((attribute) => getUrl(attribute.value))
+    .getOrElse(() => Map.of(["nothing", ""]));
+}
```

Comment:
This wasn't _quite_ what I was trying to convey, but it's an interesting take nonetheless 😄 My point was that the combination of `Source` and whatever value is associated with that `Source` might as well be _one_ string, such as `"src" + href`. You could then use that as the _key_ in a `Map`, with the values being the `<iframe>` elements that share that particular `Source` + value combination.

Either way, I realised that we've been staring ourselves blind on what to stuff in a `Set` or `Map` when the better solution is much simpler. Since we're just trying to figure out if all `<iframe>` elements definitely point to the same thing, it's just a matter of taking the first `<iframe>` element of the group and comparing it to the rest. If they're all the "same" as the first `<iframe>` then we don't need to ask a question.